test/common/tcp_proxy/BUILD

```diff
@@ -1,16 +1,19 @@
 load(
     "//bazel:envoy_build_system.bzl",
     "envoy_cc_test",
+    "envoy_cc_test_library",
     "envoy_package",
 )
 
 licenses(["notice"])  # Apache 2
 
 envoy_package()
 
-envoy_cc_test(
-    name = "tcp_proxy_test",
-    srcs = ["tcp_proxy_test.cc"],
+envoy_cc_test_library(
+    name = "tcp_proxy_test_base",
+    srcs = [
```

Comment:
I think this is compiling the .h file into a lib. Maybe this should be `hdrs` not `srcs`?

Reply:
This should be `hdrs`. Fixed and add the missing pragma once in this file.

Reply:
```
For cc_library rules, headers in hdrs comprise the public interface of the library and can be directly included both from the files in hdrs and srcs of the library itself as well as from files in hdrs and srcs of cc_* rules that list the library in their deps. Headers in srcs must only be directly included from the files in hdrs and srcs of the library itself. 
```

`tcp_proxy_base.h` should be in hdrs because it is explicitly included the `envoy_cc_test` target. It's interesting there was no error reported by bazel before the fix.